packages/core/src/Stores/common-store.js

```diff
@@ -231,6 +238,39 @@ export default class CommonStore extends BaseStore {
         this.app_routing_history.unshift(router_action);
     }
 
+    @action.bound
+    changeLanguage = (key, changeCurrentLanguage) => {
+        const request = {
+            set_settings: 1,
+            preferred_language: key,
+        };
+        SocketCache.clear();
+        if (key === 'EN') {
+            window.localStorage.setItem('i18n_language', key);
+        }
+
+        WS.setSettings(request).then(() => {
+            const new_url = new URL(window.location.href);
+            if (key === 'EN') {
+                new_url.searchParams.delete('lang');
+            } else {
+                new_url.searchParams.set('lang', key);
+            }
+            window.history.pushState({ path: new_url.toString() }, '', new_url.toString());
+            changeLanguageTranslation(key, () => {
+                changeCurrentLanguage(key);
+                BinarySocket.closeAndOpenNewConnection(key);
+            });
+        });
+    };
+
+    @action.bound
+    getExchangeRate = async (from_currency, to_currency) => {
+        const { exchange_rates } = await BinarySocket.exchange_rates(from_currency);
```

Comment:
Does it call the API everytime this function is called?

Reply:
yes, we had the same approach for `AccountSwitcher` component